### ISSUE TITLE
Trouble figuring out the correct scheme implementation

### ISSUE THREAD

**Issue by pltrz** [chriskempson/base16#182]
I am using the [st](https://st.suckless.org/) terminal emulator, and I'm trying to correctly set base16 colors.

With the help of [base16-st](https://github.com/honza/base16-st/blob/master/build/base16-hopscotch-theme.h) I get the following 16 colors set.

![2019-02-25-224913_835x205_scrot](https://user-images.githubusercontent.com/20757930/53371109-ae3ddc00-394f-11e9-89a1-9695d50989ea.png)

however using [base16-shell](https://github.com/chriskempson/base16-shell) I get the following.

![2019-02-25-224830_873x207_scrot](https://user-images.githubusercontent.com/20757930/53371069-9b2b0c00-394f-11e9-81c2-4ccf12323a26.png)

My question is: is the base16-shell how it should be and base16-st is incorrect? And should the *bright* colors be the exact same as the normal ones? Why? Should I try and get it to look like the first image but with bold colors matching normal ones or should it be like the second image?

**Comment by belak** [comment]
This is (unfortunately) simply because of how base16 is designed. There is only one set of the base colors (even though there are 16 colors total).

By default the terminal color schemes are simply a mapping of the 16 colors (really 6 colors plus shades of gray) onto the 16 terminal colors.

base16-shell attempts to remedy this by making the normal and bold versions of colors make a bit more sense but map the leftover base16 colors onto portions of the colorspace that aren't used as often so they can still be used by themes like base16-vim and base16-emacs in supported terminals.

Long story short, base16-st is "correct" even though it doesn't look great (because there's no good way to do the color mapping) and base16-shell is "correct" because it's a workaround to use all the base16 colors inside vim (and other supported applications) without having to use gui vim.

**Comment by pltrz** [comment]
Thanks for the clarification. I've resorted to just using base16-st, and modified st so that the 0-7 bold colors are the same as 0-7 normal; not sure why it does that to begin with. I don't use vim or emacs, but [vis](https://github.com/martanne/vis) instead, so I suppose I won't benefit from base16-shell and the [base16 themes for vis](https://github.com/pshevtsov/base16-vis) seem to work correctly.
